docs/proposals/all-in-one-volume.md

```diff
@@ -0,0 +1,335 @@
+includes passwords. In this case with MariaDB running, the operator may wish
+the container to have a ~/.my.cnf file that includes the username and password
+for the database.
+
+### ConfigMap + Downward API
+
+In this case, the user wishes to generate a config including the pod’s name
+(available via the metadata.name selector). This application may then pass the
+pod name along with requests in order to easily determine the source without
+using IP tracking.
+
+### Secrets + Downward API
+
+A user may wish to use a secret as a public key to encrypt the namespace of
+the pod (available via the metadata.namespace selector). This example may be
+the most contrived, but perhaps the operator wishes to use the application to
+deliver the namespace information securely without using an encrypted
+transport.
+
+### Collisions between keys when all keys from a resource are projected
+
+In the event of a user not explicitly defining keys for pathing within a volume,
+there can not be an overlap of resources with the same paths on a given volume.
+If a conflict does occur, a prefix will be added with the resource type and
+resource name. An example of this resolution is as follows:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: volume-test
+spec:
+  containers:
+  - name: container-test
+    image: busybox
+    volumeMounts:
+    - name: all-in-one
+      mountPath: "/system-volume"
+      readOnly: true
+  volumes:
+  - name: all-in-one
+    system:
+      items:
+      - secretName: mysecret
+      - configMapName: myconfigmap
+```
+
+Here the assumption is that the data values configured for each resource is the
+same. The contents of /system-volume may look like this:
+
+``/system-volume/secret-mysecret-very-generic``
+``/system-volume/configmap-myconfigmap-very-generic``
+
+### Collisions between keys when a configured path is identical
+
+In the event the user specifies any keys with the same path, the pod spec will
+not be accepted as valid. Note the specified path for mysecret and myconfigmap
+are the same:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: volume-test
+spec:
+  containers:
+  - name: container-test
+    image: busybox
+    volumeMounts:
+    - name: all-in-one
+      mountPath: "/system-volume"
+      readOnly: true
+  volumes:
+  - name: all-in-one
+    system:
+      items:
+      - secretName: mysecret
+        items:
+        - key: username
+          path: my-group/data
+      - configMapName: myconfigmap
+        items:
+        - key: config
+          path: my-group/data
+```
+
+### Collisions when one path is explicit and the other is automatically projected
+
+In the event that there is a collision due to a user specified path matching
+data that is automatically projected, resolution will be done using the same
+numeric prefix addition as described in the first scenario.
+
+## Code changes
+
+### Proposed API objects
+
+```go
+type SystemVolumeSource struct {
```

Comment:
This name is bad.  It doesn't convey any information.

"Projections" or something like that would be better